server/src/test/java/org/elasticsearch/action/explain/ExplainResponseTests.java

```diff
@@ -0,0 +1,131 @@
+    @Override
+    protected ExplainResponse doParseInstance(XContentParser parser) throws IOException {
+        return ExplainResponse.fromXContent(parser, randomBoolean());
+    }
+
+    @Override
+    protected ExplainResponse createBlankInstance() {
+        return new ExplainResponse();
+    }
+
+    @Override
+    protected ExplainResponse createTestInstance() {
+        String index = randomAlphaOfLength(5);
+        String type = randomAlphaOfLength(5);
+        String id = String.valueOf(randomIntBetween(1,100));
+        boolean exist = randomBoolean();
+        Explanation explanation = randomExplanation(randomExplanation(randomExplanation()), randomExplanation());
+        String fieldName = randomAlphaOfLength(10);
+        List<Object> values = Arrays.asList(randomAlphaOfLengthBetween(3, 10), randomInt(), randomLong(), randomDouble(), randomBoolean());
+        GetResult getResult = new GetResult(randomAlphaOfLengthBetween(3, 10),
+            randomAlphaOfLengthBetween(3, 10),
+            randomAlphaOfLengthBetween(3, 10),
+            randomNonNegativeLong(),
+            true,
+            RandomObjects.randomSource(random()),
+            singletonMap(fieldName, new DocumentField(fieldName, values)));
+        return new ExplainResponse(index, type, id, exist, explanation, getResult);
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return field -> field.startsWith("fields") || field.startsWith("get");
+    }
+
+    public void testToXContent() throws IOException {
+        ExplainResponse response = createSimpleResponse();
+
+        XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+        response.toXContent(builder, ToXContent.EMPTY_PARAMS);
+
+        String generatedResponse = BytesReference.bytes(builder).utf8ToString().replaceAll("\\s+", "");
+
+        String expectedResponse =
+            ("{\n" +
+            "    \"_index\":\"index\",\n" +
+            "    \"_type\":\"type\",\n" +
+            "    \"_id\":\"1\",\n" +
+            "    \"matched\":true,\n" +
+            "    \"explanation\":{\n" +
+            "        \"value\":1.0,\n" +
+            "        \"description\":\"description\",\n" +
+            "        \"details\":[]\n" +
+            "    },\n" +
+            "    \"get\":{\n" +
+            "        \"found\":true,\n" +
+            "        \"_source\":{\n" +
+            "            \"field1\":\"value1\",\n" +
+            "            \"field2\":\"value2\"\n" +
+            "        },\n" +
+            "        \"fields\":{\n" +
+            "            \"field1\":[\n" +
+            "                \"value1\"\n" +
+            "            ]\n" +
+            "        }\n" +
+            "    }\n" +
+            "}").replaceAll("\\s+", "");
+        assertThat(expectedResponse, equalTo(generatedResponse));
+    }
+
+    private ExplainResponse createSimpleResponse() {
```

Comment:
given that we use this in a single place, can we remove this method and put the code where it's currently called?